ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -552,10 +552,21 @@ public func mergeMap<T, U>(transform: T -> SignalProducer<U>)(producer: SignalPr
 public func switch<T>(producer: SignalProducer<SignalProducer<T>>) -> SignalProducer<T>
 public func switchMap<T, U>(transform: T -> SignalProducer<U>)(producer: SignalProducer<T>) -> SignalProducer<U>
 
-public func repeat<T>(count: Int)(producer: SignalProducer<T>) -> SignalProducer<T>
 public func retry<T>(count: Int)(producer: SignalProducer<T>) -> SignalProducer<T>
 */
 
+/// Repeats `producer` `count` times.
+/// Repeating `0` times results in a equivalent signal producer.
+public func repeat<T, E>(count: Int)(producer: SignalProducer<T, E>) -> SignalProducer<T, E> {
+	precondition(count >= 0)
```

Comment:
A `precondition` like this is fine.

This actually raises an interesting question, though. My mental model of `count` was “the number of times you want to see the signal events,” but you've interpreted it as “the number of _additional_ times.”

The problem is that I don't think either interpretation is actually wrong. Maybe we should call this operator `times()` or something instead, so that it's clear which semantics it has?